src/os/ui/data/adddatactrl.js

```diff
@@ -214,13 +214,40 @@ os.ui.data.AddDataCtrl.listFilter_ = function(item, i, arr) {
     }
 
     // exclude providers without children so users don't think they can do something with them (unless flagged)
-    var children = item.getChildren();
-    return item.getShowWhenEmpty() || (!!children && children.length > 0);
+    if (os.ui.data.AddDataCtrl.itemHasChildren_(item)) {
+      return true;
+    } else {
+      return os.ui.data.AddDataCtrl.showWhenEmpty_(item);
+    }
   }
 
   return false;
 };
 
+/**
+ * Check if the item should be shown even if empty.
+ *
+ * @param {os.ui.slick.SlickTreeNode} item
+ * @return {boolean} if should be shown even when empty
+ */
+os.ui.data.AddDataCtrl.showWhenEmpty_ = function(item) {
+  if (os.implements(item, os.data.IDataProvider.ID)) {
```

Comment:
Everything that implements `os.data.IDataProvider` is going to need to register with the implement method for this to recognize if `item` meets the qualifications.
For example `ogcserver.js` is going to need this line after the constructor:  `os.implements(os.ui.ogc.OGCServer, os.data.IDataDescriptor.ID);` Another example of how a class registers with the implements method: https://github.com/ngageoint/opensphere/blob/master/src/os/data/basedescriptor.js#L156
Looks like 7 files are directly affected, plus `abstractloadingserver.js` should get it since `os.ui.server.ILoadingProvider` extends `os.data.IDataProvider`

Reply:
Presumably that should be `os.implements(os.ui.ogc.OGCServer, os.data.IDataProvider.ID);`?

Reply:
Ah right, I was looking for the wrong thing. You have the right idea there